### PR TITLE
SOV-3531: Refactor charts

### PR DESCRIPTION
**Pull request by tiltom**
Resolves https://sovryn.atlassian.net/browse/SOV-3531

### CODE REVIEW

File: apps/frontend/src/app/2_molecules/Chart/Chart.utils.tsx [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+  yLabel1: string,
+  yLabel2: string,
+) => {
+  const primaryColor = '#2C303B';
+  const textColor = '#f5f5f5';
+  const borderColor = '#484D59';
+  const fontFamily = 'Roboto';
+  const fontSize = 12;
+  const fontWeight = '500';
+  const dashGridType = 'dash';
```
Comment (Rickk137) [inline apps/frontend/src/app/2_molecules/Chart/Chart.utils.tsx]: You can move these constant into `Chart.constant.tsx`

File: apps/frontend/src/app/2_molecules/Chart/Chart.utils.tsx [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        data: mockData.data2,
+        backgroundColor: gradient2,
+        borderColor: mockData.borderColor2,
+        fill: true,
+        yAxisID: 'y',
+        pointRadius: 0,
+      },
+    ],
+  };
+};
```
Comment (Rickk137) [inline apps/frontend/src/app/2_molecules/Chart/Chart.utils.tsx]: `) => ({
    labels: mockData.xLabels,
    datasets: [
      {
        label: mockData.label1,
        data: mockData.data1,
        backgroundColor: gradient1,
        borderColor: mockData.borderColor1,
        fill: true,
        yAxisID: 'y1',
        pointRadius: 0,
      },
      {
        label: mockData.label2,
        data: mockData.data2,
        backgroundColor: gradient2,
        borderColor: mockData.borderColor2,
        fill: true,
        yAxisID: 'y',
        pointRadius: 0,
      },
    ],
  };)`

File: apps/frontend/src/app/2_molecules/Chart/Chart.utils.tsx [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+  id: 'customCanvasBackgroundColor',
+  beforeDraw: (chart, options) => {
+    const { ctx } = chart;
+    ctx.save();
+    ctx.globalCompositeOperation = 'destination-over';
+    ctx.fillStyle = options.color || '#2C303B';
+    ctx.fillRect(0, 0, chart.width, chart.height);
+    ctx.restore();
+  },
+};
```
Comment (Rickk137) [inline apps/frontend/src/app/2_molecules/Chart/Chart.utils.tsx]: this is a constant

File: apps/frontend/src/app/5_pages/LendPage/components/LendFrame/components/LendFrameChart/LendFrameChart.utils.tsx [hunk 4]
```diff
@@ -16,179 +15,12 @@ export const convertPoolHistoryToMockData = (
...
-    animations: {
-      tension: {
-        duration: 1000,
-        from: 1,
-        to: 0,
-      },
-    },
+    data1: lendApr,
+    label1: t(translations.lendPage.table.lendApr),
+    borderColor1: '#72EADE',
```
Comment (Rickk137) [inline apps/frontend/src/app/5_pages/LendPage/components/LendFrame/components/LendFrameChart/LendFrameChart.utils.tsx]: can you use the color `positive` from tailwind?

File: apps/frontend/src/app/5_pages/LendPage/components/LendFrame/components/LendFrameChart/LendFrameChart.utils.tsx [hunk 5]
```diff
@@ -16,179 +15,12 @@ export const convertPoolHistoryToMockData = (
...
+    data2: totalLiquidity,
+    label2: t(translations.lendPage.table.totalLiquidity),
+    borderColor2: '#82868F',
```
Comment (Rickk137) [inline apps/frontend/src/app/5_pages/LendPage/components/LendFrame/components/LendFrameChart/LendFrameChart.utils.tsx]: can you use the color `gray-40` from tailwind?

File: apps/frontend/src/app/5_pages/LendPage/components/LendFrame/components/LendFrameChart/LendFrameChart.tsx [hunk 6]
```diff
@@ -1,82 +1,34 @@
...
-    </div>
+    <Chart
+      mockData={mockData}
+      tickStep={tickStep}
+      yLabel1={t(translations.lendPage.table.lendApr)}
+      yLabel2={`${t(
+        translations.lendPage.table.totalLiquidity,
+      )} ${getTokenDisplayName(pool.getAsset())}`}
+      gradient1Colors={['rgba(114, 234, 222, 1)', 'rgba(114, 234, 222, 0.09']}
+      gradient2Colors={['rgba(130, 134, 143, 1)', 'rgba(130, 134, 143, 0.09)']}
```
Comment (Rickk137) [inline apps/frontend/src/app/5_pages/LendPage/components/LendFrame/components/LendFrameChart/LendFrameChart.tsx]: define these gradients in a constant and use it

File: apps/frontend/src/app/5_pages/MarketMakingPage/components/PoolDetails/components/PoolChart/PoolChart.tsx [hunk 7]
```diff
@@ -1,87 +1,34 @@
...
   return (
-    <div className="lg:h-[37rem] h-64 rounded">
-      <canvas ref={canvas}></canvas>
-    </div>
+    <Chart
+      mockData={mockData}
+      tickStep={tickStep}
+      yLabel1={t(translations.marketMakingPage.poolVolumeChart.apr)}
+      yLabel2={t(translations.marketMakingPage.poolVolumeChart.volume)}
+      gradient1Colors={['rgba(130, 134, 143, 1)', 'rgba(130, 134, 143, 0.09)']}
```
Comment (Rickk137) [inline apps/frontend/src/app/5_pages/MarketMakingPage/components/PoolDetails/components/PoolChart/PoolChart.tsx]: define these gradients in a constant and use it

**Review by Rickk137 — CHANGES_REQUESTED** [review]

**Review by rick23p — APPROVED** [review]

**Pull request merged**